### PR TITLE
add package tidy-html5

### PR DESCRIPTION
**Pull request by tehnick**

### CODE REVIEW

**Comment by tehnick** [comment]
Rebased onto current master.

Any comments?

File: src/tidy-html5.mk [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+define $(PKG)_UPDATE
+    $(WGET) -q -O- 'https://github.com/htacg/tidy-html5/releases' | \
+    $(SED) -n 's,.*/htacg/tidy-html5/archive/\([0-9][^>]*\)\.tar\.gz.*,\1,p' | \
+    head -1
+endef
+
+define $(PKG)_BUILD
+    mkdir '$(1).build'
+    cd    '$(1).build' && $(TARGET)-cmake '$(1)' \
```
Comment (starius) [inline src/tidy-html5.mk]: Please use BUILD_DIT and SOURCE_DIR as in https://github.com/mxe/mxe/pull/1709#issuecomment-286680056

File: src/tidy-html5.mk [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        -DTIDY_COMPAT_HEADERS:BOOL=YES \
+        -DBUILD_SHARED_LIB=$(CMAKE_SHARED_BOOL)
+    $(MAKE) -C '$(1).build' -j '$(JOBS)'
+    $(MAKE) -C '$(1).build' -j 1 install
+
+    $(if $(BUILD_STATIC),
+        cd '$(PREFIX)/$(TARGET)/lib' && mv libtidys.a libtidy.a,
+        rm -f '$(PREFIX)/$(TARGET)/lib/libtidys.a')
+endef
```
Comment (starius) [inline src/tidy-html5.mk]: Please add a test.

File: src/tidy-html5.mk [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+PKG             := tidy-html5
+$(PKG)_WEBSITE  := http://www.html-tidy.org/
+$(PKG)_DESCR    := HTML/XML syntax checker and reformatter
+$(PKG)_IGNORE   :=
+$(PKG)_VERSION  := 5.4.0
+$(PKG)_CHECKSUM := a2d754b7349982e33f12d798780316c047a3b264240dc6bbd4641542e57a0b7a
+$(PKG)_SUBDIR   := $(PKG)-$($(PKG)_VERSION)
+$(PKG)_FILE     := $(PKG)-$($(PKG)_VERSION).tar.gz
+$(PKG)_URL      := https://github.com/htacg/tidy-html5/archive/$($(PKG)_VERSION).tar.gz
```
Comment (starius) [inline src/tidy-html5.mk]: After #1712, you can replace $(PKG)_FILE, $(PKG)_SUBDIR, $(PKG)_URL, and $(PKG)_UPDATE with:

$(PKG)_GH_CONF

Comment (tehnick) [inline src/tidy-html5.mk]: Done.

File: src/tidy-html5.mk (hunk 1 above)
Comment (tehnick) [inline src/tidy-html5.mk]: Done.

File: src/tidy-html5.mk (hunk 2 above)
Comment (tehnick) [inline src/tidy-html5.mk]: Maybe I'll add it later.

Is it prevents from merging?

Comment (tonytheodore) [inline src/tidy-html5.mk]: See https://github.com/mxe/mxe/pull/1709#issuecomment-289258977, the best time to add a test is when the package is new and fresh in your mind.

**Comment by tehnick** [comment]
I have added a test.

**Comment by starius** [comment]
Tested on all 4 targets. It installs files `usr/<target>bin/tidy.exe`. Can it be avoided?
Otherwise it is ready to go.

**Comment by tehnick** [comment]
> It installs files usr/<target>bin/tidy.exe. Can it be avoided?

Via configuration options -- no.

It may be removed only manually. See the last commit.

**Comment by starius** [comment]
The does the test name contain "pkgconfig" if `pkgconfig` is not used to build it?

File: src/tidy-html5-test.c [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        printf("Unknown error: %d.\n\n", rc);
+    fflush(stdout);
+
+    tidyBufFree(&errBuf);
+    tidyBufFree(&output);
+    tidyRelease(tDoc);
+
+    return rc;
+}
+
```
Comment (starius) [inline src/tidy-html5-test.c]: Trailing empty line

**Comment by tehnick** [comment]
Updated.

**Comment by starius** [comment]
Looks good to me!
Thanks!

**Pull request merged**